src/rsz/src/Rebuffer.cc

```diff
@@ -239,14 +239,27 @@ RepairSetup::rebufferBottomUp(BufferedNetPtr bnet,
     // larger required and smaller capacitance.
     // This is fanout^2.
     // Presort options to hit better options sooner.
-    sort(Z.begin(), Z.end(),
-         [=](BufferedNetPtr option1,
-             BufferedNetPtr option2)
-         { Slack slack1 = slackPenalized(option1);
+    sort(Z.begin(),
+         Z.end(),
+         [=](BufferedNetPtr option1, BufferedNetPtr option2) {
+           Slack slack1 = slackPenalized(option1);
            Slack slack2 = slackPenalized(option2);
-           return fuzzyGreater(slack1, slack2)
-             || (fuzzyEqual(slack1, slack2)
-                 && fuzzyLess(option1->cap(), option2->cap()));
+
+           if (slack1 > slack2) {
+             return true;
+           }
```

Comment:
This is not semantically equivalent - you dropped all the fuzzy-ness from the comparisons.

Reply:
You cannot have fuzzy-ness in comparators.

One of the properties is transitivity of equivalence.

if equiv(a,b) and equiv(b,c) then equiv (a,c) is true. The following example demonstrates a the problem with the fuzzy equiv comparators which is fundamentally what STA does.

```
struct MyComparator {
  bool operator()(double a, double b) const {
    if (std::abs(a - b) <= 1.0) {
      return false;  // Close-enough values are considered equal.
    }
    return a < b;
  }
};

TEST(MyComparator, HasAProblem) {
  std::set<double, MyComparator> s1 = gtl::Container(1, 2, 3);
  std::set<double, MyComparator> s2 = gtl::Container(2, 3, 1);
  EXPECT_EQ(s1.size(), s2.size());
  // The above fails: s1.size() == 2, but s2.size() == 1.
}
```

Reply:
In the example above comp(1,2) is equal and comp(2,3) is equal, but comp(1,3) is not equal.

Reply:
Ok but this will need a full CI testing as it changes the semantics (for the better but even so).

Reply:
Sounds good. This one is high priority if you can make sure to run it soon.

Reply:
running